From 62b8484e347d0c501175cca43db476615c6f3cd4 Mon Sep 17 00:00:00 2001
From: Gabriel Wicki <gabriel.wicki@bytesatwork.ch>
Date: Thu, 21 Apr 2022 13:38:26 +0200
Subject: [PATCH] arm: dts: stm32mp157c-bytedevkit: Add support for byteDEVKIT
 V1.3

Enable EEPROM support so that byteDEVKIT V1.3 can be detected and the right
kernel and device tree can be loaded.

Note that by default, U-Boot for byteDEVKIT V1.3 is built.

Signed-off-by: Gabriel Wicki <gabriel.wicki@bytesatwork.ch>
---
 arch/arm/dts/Makefile                         |  3 +-
 .../stm32mp157c-bytedevkit-v1-1-u-boot.dtsi   |  1 +
 arch/arm/dts/stm32mp157c-bytedevkit-v1-1.dts  |  8 +++++
 ...> stm32mp157c-bytedevkit-v1-3-u-boot.dtsi} |  1 +
 arch/arm/dts/stm32mp157c-bytedevkit-v1-3.dts  | 14 +++++++++
 ...devkit.dts => stm32mp157c-bytedevkit.dtsi} |  2 --
 arch/arm/dts/stm32mp157c-byteengine-m5.dtsi   | 29 +++++++++++++++++++
 configs/stm32mp157_bytedevkit_1g_defconfig    |  9 +++++-
 configs/stm32mp157_bytedevkit_defconfig       |  9 +++++-
 9 files changed, 71 insertions(+), 5 deletions(-)
 create mode 120000 arch/arm/dts/stm32mp157c-bytedevkit-v1-1-u-boot.dtsi
 create mode 100644 arch/arm/dts/stm32mp157c-bytedevkit-v1-1.dts
 rename arch/arm/dts/{stm32mp157c-bytedevkit-u-boot.dtsi => stm32mp157c-bytedevkit-v1-3-u-boot.dtsi} (99%)
 create mode 100644 arch/arm/dts/stm32mp157c-bytedevkit-v1-3.dts
 rename arch/arm/dts/{stm32mp157c-bytedevkit.dts => stm32mp157c-bytedevkit.dtsi} (99%)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index a91c19244b..c4b9d885bb 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -837,7 +837,8 @@ dtb-$(CONFIG_STM32MP15x) += \
 	stm32mp157f-dk2.dtb \
 	stm32mp157f-ed1.dtb \
 	stm32mp157f-ev1.dtb \
-	stm32mp157c-bytedevkit.dtb \
+	stm32mp157c-bytedevkit-v1-1.dtb \
+	stm32mp157c-bytedevkit-v1-3.dtb \
 	stm32mp15xx-dhcom-pdk2.dtb
 
 dtb-$(CONFIG_SOC_K3_AM6) += k3-am654-base-board.dtb k3-am654-r5-base-board.dtb
diff --git a/arch/arm/dts/stm32mp157c-bytedevkit-v1-1-u-boot.dtsi b/arch/arm/dts/stm32mp157c-bytedevkit-v1-1-u-boot.dtsi
new file mode 120000
index 0000000000..710f42c394
--- /dev/null
+++ b/arch/arm/dts/stm32mp157c-bytedevkit-v1-1-u-boot.dtsi
@@ -0,0 +1 @@
+stm32mp157c-bytedevkit-v1-3-u-boot.dtsi
\ No newline at end of file
diff --git a/arch/arm/dts/stm32mp157c-bytedevkit-v1-1.dts b/arch/arm/dts/stm32mp157c-bytedevkit-v1-1.dts
new file mode 100644
index 0000000000..6caaae9166
--- /dev/null
+++ b/arch/arm/dts/stm32mp157c-bytedevkit-v1-1.dts
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ *  Copyright (C) 2022 bytes at work AG - http://www.bytesatwork.ch
+ */
+
+/dts-v1/;
+
+#include "stm32mp157c-bytedevkit.dtsi"
diff --git a/arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi b/arch/arm/dts/stm32mp157c-bytedevkit-v1-3-u-boot.dtsi
similarity index 99%
rename from arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi
rename to arch/arm/dts/stm32mp157c-bytedevkit-v1-3-u-boot.dtsi
index 0ed92e982d..f5aabfd1df 100644
--- a/arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi
+++ b/arch/arm/dts/stm32mp157c-bytedevkit-v1-3-u-boot.dtsi
@@ -16,6 +16,7 @@
 
 / {
 	aliases {
+		i2c1 = &i2c2;
 		i2c3 = &i2c4;
 		mmc0 = &sdmmc1;
 		spi0 = &qspi;
diff --git a/arch/arm/dts/stm32mp157c-bytedevkit-v1-3.dts b/arch/arm/dts/stm32mp157c-bytedevkit-v1-3.dts
new file mode 100644
index 0000000000..84e69da7a3
--- /dev/null
+++ b/arch/arm/dts/stm32mp157c-bytedevkit-v1-3.dts
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ *  Copyright (C) 2022 bytes at work AG - http://www.bytesatwork.ch
+ */
+
+/dts-v1/;
+
+#include "stm32mp157c-bytedevkit.dtsi"
+
+&i2c2 {
+	eeprom@57 {
+		reg = <0x57>;
+	};
+};
diff --git a/arch/arm/dts/stm32mp157c-bytedevkit.dts b/arch/arm/dts/stm32mp157c-bytedevkit.dtsi
similarity index 99%
rename from arch/arm/dts/stm32mp157c-bytedevkit.dts
rename to arch/arm/dts/stm32mp157c-bytedevkit.dtsi
index 7e47a9e689..034f584715 100644
--- a/arch/arm/dts/stm32mp157c-bytedevkit.dts
+++ b/arch/arm/dts/stm32mp157c-bytedevkit.dtsi
@@ -3,8 +3,6 @@
  *  Copyright (C) 2019 bytes at work AG - http://www.bytesatwork.ch
  */
 
-/dts-v1/;
-
 #include "stm32mp157c-byteengine-m5.dtsi"
 
 / {
diff --git a/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi b/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi
index b0ad337709..4eaa714d2d 100644
--- a/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi
+++ b/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi
@@ -75,6 +75,23 @@
 };
 
 &pinctrl {
+	i2c2_pins: i2c2-0 {
+		pins {
+			pinmux = <STM32_PINMUX('H', 4, AF4)>, /* I2C2_SCL */
+				 <STM32_PINMUX('H', 5, AF4)>; /* IC2C_SDA */
+			bias-pull-up;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+
+	i2c2_pins_sleep: i2c2-1 {
+		pins {
+			pinmux = <STM32_PINMUX('H', 4, ANALOG)>, /* I2C2_SCL */
+				 <STM32_PINMUX('H', 5, ANALOG)>; /* I2C2_SDA */
+		};
+	};
+
 	sdmmc2_pins: sdmmc2-b4-0 {
 		pins1 {
 			pinmux = <STM32_PINMUX('B', 14, AF9)>, /* SDMMC2_D0 */
@@ -159,6 +176,18 @@
 	};
 };
 
+/* HWID EEPROM Configuration */
+&i2c2 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&i2c2_pins>;
+	pinctrl-1 = <&i2c2_pins_sleep>;
+	status = "okay";
+
+	hwid: eeprom@57 {
+		reg = <0x57>;
+	};
+};
+
 /* PMIC and bytesatwork configuration EEPROM */
 &i2c4 {
 	pinctrl-names = "default", "sleep";
diff --git a/configs/stm32mp157_bytedevkit_1g_defconfig b/configs/stm32mp157_bytedevkit_1g_defconfig
index 713857f10a..1fe3ff190e 100644
--- a/configs/stm32mp157_bytedevkit_1g_defconfig
+++ b/configs/stm32mp157_bytedevkit_1g_defconfig
@@ -29,6 +29,7 @@ CONFIG_CMD_BOOTZ=y
 # CONFIG_CMD_XIMG is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
+CONFIG_CMD_EEPROM=y
 CONFIG_CMD_MEMINFO=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_ADC=y
@@ -61,7 +62,7 @@ CONFIG_MTDIDS_DEFAULT="nor0=w25q128-flash.0"
 CONFIG_MTDPARTS_DEFAULT="mtdparts_nor1=256k(fsbl1),256k(fsbl2),2m(ssbl),256k(u-boot-env),-(nor_user)"
 # CONFIG_SPL_DOS_PARTITION is not set
 CONFIG_ISO_PARTITION=y
-CONFIG_DEFAULT_DEVICE_TREE="stm32mp157c-bytedevkit"
+CONFIG_DEFAULT_DEVICE_TREE="stm32mp157c-bytedevkit-v1-3"
 CONFIG_OF_SPL_REMOVE_PROPS="interrupt-parent"
 CONFIG_DM_HWSPINLOCK=y
 CONFIG_HWSPINLOCK_STM32=y
@@ -69,6 +70,12 @@ CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_STM32F7=y
 CONFIG_DM_MAILBOX=y
 CONFIG_STM32_IPCC=y
+CONFIG_I2C_EEPROM=y
+CONFIG_SYS_I2C_EEPROM_ADDR=0x57
+CONFIG_SYS_I2C_EEPROM_BUS=1
+CONFIG_SYS_EEPROM_SIZE=32768
+CONFIG_SYS_EEPROM_PAGE_WRITE_BITS=8
+CONFIG_SYS_I2C_EEPROM_ADDR_LEN=2
 CONFIG_DM_MMC=y
 CONFIG_STM32_SDMMC2=y
 CONFIG_MTD=y
diff --git a/configs/stm32mp157_bytedevkit_defconfig b/configs/stm32mp157_bytedevkit_defconfig
index 9b52ea7d6b..315ffc7db7 100644
--- a/configs/stm32mp157_bytedevkit_defconfig
+++ b/configs/stm32mp157_bytedevkit_defconfig
@@ -28,6 +28,7 @@ CONFIG_CMD_BOOTZ=y
 # CONFIG_CMD_XIMG is not set
 # CONFIG_CMD_EXPORTENV is not set
 # CONFIG_CMD_IMPORTENV is not set
+CONFIG_CMD_EEPROM=y
 CONFIG_CMD_MEMINFO=y
 CONFIG_CMD_MEMTEST=y
 CONFIG_CMD_ADC=y
@@ -60,7 +61,7 @@ CONFIG_MTDIDS_DEFAULT="nor0=w25q128-flash.0"
 CONFIG_MTDPARTS_DEFAULT="mtdparts_nor1=256k(fsbl1),256k(fsbl2),2m(ssbl),256k(u-boot-env),-(nor_user)"
 # CONFIG_SPL_DOS_PARTITION is not set
 CONFIG_ISO_PARTITION=y
-CONFIG_DEFAULT_DEVICE_TREE="stm32mp157c-bytedevkit"
+CONFIG_DEFAULT_DEVICE_TREE="stm32mp157c-bytedevkit-v1-3"
 CONFIG_OF_SPL_REMOVE_PROPS="interrupt-parent"
 CONFIG_DM_HWSPINLOCK=y
 CONFIG_HWSPINLOCK_STM32=y
@@ -68,6 +69,12 @@ CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_STM32F7=y
 CONFIG_DM_MAILBOX=y
 CONFIG_STM32_IPCC=y
+CONFIG_I2C_EEPROM=y
+CONFIG_SYS_I2C_EEPROM_ADDR=0x57
+CONFIG_SYS_I2C_EEPROM_BUS=1
+CONFIG_SYS_EEPROM_SIZE=32768
+CONFIG_SYS_EEPROM_PAGE_WRITE_BITS=8
+CONFIG_SYS_I2C_EEPROM_ADDR_LEN=2
 CONFIG_DM_MMC=y
 CONFIG_STM32_SDMMC2=y
 CONFIG_MTD=y
-- 
2.20.1

